src/bin/rahmen.rs

```diff
@@ -158,7 +159,18 @@ fn main() -> RahmenResult<()> {
         eprintln!("Config file not found, continuing with default settings");
         Default::default()
     };
-    let status_line_formatter = StatusLineFormatter::new(settings.status_line.iter().cloned())?;
+    // build the status line, using the settings from the config file (first for the individual
+    // metadata tags, second for the regex(es) to process the whole status line),
+    // the metadata items being joined using the separator from the config file (or with nothing
+    // if no separator is given there)
+    let status_line_formatter = StatusLineFormatter::new(
+        settings.status_line.iter().cloned(),
+        settings.line_replace.iter().flat_map(identity).cloned(),
+        // TODO I do not like this hack. It should evaluate as None if none is given in config,
+        // and we should process it as None, i.e. do not call join in the formatter.
+        // How can this be done without having an unwieldy if construction with four branches?
+        settings.separator.unwrap_or("".to_string()),
```

Comment:
Not quite sure I understand this part. I currently don't see a use case where the separator would be `None` and it would still produce output. Why not have a sensible default (e.g., `", "`) and let the user override it?

Reply:
It's OK now, I changed the whole mechanism.